### PR TITLE
Adding augroup, auexcludegroup, aurights support for allUsers.

### PR DESCRIPTION
**Pull request by Cladis**
Adding augroup, auexcludegroup, aurights support for allUsers. Adding functions calling just them. Fixing "-1" value for "number" proceeding (as is used in allUsersWithPrefix originally).

### CODE REVIEW

File: src/org/wikipedia/Wiki.java [hunk 1]
```diff
@@ -4032,12 +4068,15 @@ public boolean userExists(String username) throws IOException
      *  @param start the string to start enumeration
      *  @param number the number of users to return
      *  @param prefix list all users with this prefix (overrides start and amount),
-     *  use "" to not specify one
+     *  @param group list all users in this group(s). Use pipe-char "|" to separate group names.
+     *  @param excludegroup list all users who are not in this group(s). Use pipe-char "|" to separate group names.
+     *  @param rights list all users with this right(s). Use pipe-char "|" to separate right names.
+     *  use "" to not
```
Comment (MER-C) [inline src/org/wikipedia/Wiki.java]: You'll need to shuffle this around a little.

File: src/org/wikipedia/Wiki.java [hunk 2]
```diff
@@ -4010,10 +4010,46 @@ public boolean userExists(String username) throws IOException
...
-        return allUsers(start, number, "");
+        return allUsers(start, number, "", "", "", "");
     }
 
     /**
+     *  Returns all usernames of users in given group(s)
+     *  @param group a group name. Use pipe-char "|" to separate if several 
+     *  @return (see above)
+     *  @throws IOException if a network error occurs
+     *  @since 0.28
```
Comment (MER-C) [inline src/org/wikipedia/Wiki.java]: Please update to 0.32.

**Comment by MER-C** [comment]
Please add unit tests for: nonsense string for allUsersInGroup and allUsersWithRight (should return empty array).

**Comment by Cladis** [comment]
Sorry I'm not a programmer and have never performed unit testing thus a very stupid question: are you asking to add another two or so lines to /test/org/wikipedia/WikiUnitTest.java ?

**Comment by MER-C** [comment]
I wasn't clear enough: the documentation comment should be
```
      *  @param prefix list all users with this prefix (overrides start and amount), 
      *  use "" to not specify one 
```
You split this by inserting the other parameters by inserting the other parameters between the two lines.

> are you asking to add another two or so lines to /test/org/wikipedia/WikiUnitTest.java

Yes. Shouldn't be too difficult, just copy what's already there. Something like the following will do:

```java
@Test
public void allUsersInGroup() throws IOException
{
    assertEquals("allUsersInGroup: nonsense", new String[0], testWiki.allUsersInGroup("sdfkd|&");
}
```

Also, please fixup your commits (`git rebase --interactive 76ecc35`).

**Comment by Cladis** [comment]
About the documentation: ah, but the new params also have empty string as a way to not specify one. I can't think of a way to make it so that it looks good :(

**Comment by MER-C** [comment]
I would repeat that phrase for each parameter that it is applicable for. When I run javadoc, it will come out something like [this](https://wikipediatools.appspot.com/doc/org/wikipedia/Wiki.html#allUsers-java.lang.String-int-java.lang.String-).

**Comment by MER-C** [comment]
Fixed up and merged.
